src/detect-file-data.c

```diff
@@ -84,7 +84,7 @@ static int DetectFiledataSetup (DetectEngineCtx *de_ctx, Signature *s, char *str
         return -1;
     }
 
-    if ((s->init_flags & SIG_FLAG_INIT_FLOW) &&
+    if (s->alproto == ALPROTO_HTTP && (s->init_flags & SIG_FLAG_INIT_FLOW) &&
```

Comment:
On 06/05/2015 10:31 AM, Giuseppe Longo wrote:
> In src/detect-file-data.c
> <https://github.com/inliniac/suricata/pull/1534#discussion_r31795476>:
> 
>> @@ -84,7 +84,7 @@ static int DetectFiledataSetup (DetectEngineCtx *de_ctx, Signature *s, char *str
>>          return -1;
>>      }
>>  
>> -    if ((s->init_flags & SIG_FLAG_INIT_FLOW) &&
>> +    if (s->alproto == ALPROTO_HTTP && (s->init_flags & SIG_FLAG_INIT_FLOW) &&
> 
> ok, I see. thanks

This is also an indication of insufficient testing, so make sure you add
tests that deal with both valid and invalid rules.